packages/utils/src/dappkit.ts

```diff
@@ -0,0 +1,178 @@
+  type: DappKitRequestTypes.ACCOUNT_ADDRESS
+  status: DappKitResponseStatus.UNAUTHORIZED
+}
+
+export type AccountAuthResponse = AccountAuthResponseSuccess | AccountAuthResponseFailure
+
+export interface SignTxResponseSuccess {
+  type: DappKitRequestTypes.SIGN_TX
+  status: DappKitResponseStatus.SUCCESS
+  rawTx: string
+}
+
+export const SignTxResponseSuccess = (rawTx: string): SignTxResponseSuccess => ({
+  type: DappKitRequestTypes.SIGN_TX,
+  status: DappKitResponseStatus.SUCCESS,
+  rawTx,
+})
+
+export interface SignTxResponseFailure {
+  type: DappKitRequestTypes.SIGN_TX
+  status: DappKitResponseStatus.UNAUTHORIZED
+}
+
+export type SignTxResponse = SignTxResponseSuccess | SignTxResponseFailure
+
+export type DappKitResponse = AccountAuthResponse | SignTxResponse
+
+export function produceResponseDeeplink(request: DappKitRequest, response: DappKitResponse) {
+  let params: any = { type: response.type, status: response.status, requestId: request.requestId }
+  switch (response.type) {
+    case DappKitRequestTypes.ACCOUNT_ADDRESS:
+      if (response.status === DappKitResponseStatus.SUCCESS) {
+        params.account = response.address
+      }
+      break
+    case DappKitRequestTypes.SIGN_TX:
+      if (response.status === DappKitResponseStatus.SUCCESS) {
+        params.rawTx = response.rawTx
+      }
+    default:
+      break
+  }
+
+  return request.callback + '?' + stringify(params)
+}
+
+export interface SignTxRequest extends DappKitRequestBase {
+  type: DappKitRequestTypes.SIGN_TX
+  txData: string
+  estimatedGas: number
+  from: string
+  to: string
+  nonce: number
+  gasCurrencyAddress: string
+}
+
+export const SignTxRequest = (
+  txData: string,
+  estimatedGas: number,
+  from: string,
+  to: string,
+  nonce: number,
+  gasCurrencyAddress: string,
+  callback: string,
+  requestId: string
+): SignTxRequest => ({
+  type: DappKitRequestTypes.SIGN_TX,
+  txData,
+  estimatedGas,
+  from,
+  to,
+  nonce,
+  gasCurrencyAddress,
+  callback,
+  requestId,
+})
+
+export type DappKitRequest = AccountAuthRequest | SignTxRequest
+
+export function serializeDappKitRequestDeeplink(request: DappKitRequest) {
+  let params: any = { type: request.type, requestId: request.requestId, callback: request.callback }
+  switch (request.type) {
+    case DappKitRequestTypes.SIGN_TX:
+      params = {
+        ...params,
+        txData: request.txData,
+        estimatedGas: request.estimatedGas,
+        from: request.from,
+        to: request.to,
+        nonce: request.nonce,
+        gasCurrencyAddress: request.gasCurrencyAddress,
+      }
+    default:
+      break
+  }
+
+  return DAPPKIT_BASE_HOST + '?' + stringify(params)
+}
+
+export function parseDappKitRequestDeeplink(url: string): DappKitRequest {
+  const rawParams = parse(url, true)
+
+  if (rawParams.query.type === undefined) {
+    throw new Error('Invalid Deeplink: does not contain type' + url)
+  }
+  switch (rawParams.query.type) {
```

Comment:
Why all the ts ignores here?

Reply:
`rawParams.query` values are all `string | string[]` which cause a mismatch. Once things stablize the parsing should properly validate the payload etc., but figure it would be premature while the interface is influx

Reply:
Okay, can you leave a TODO to this effect?